### PR TITLE
bug/SWTCH-1401 incorrect expiry

### PR DESCRIPTION
**Pull request by JGiter**
### Summary

|             |   |
|-------------|---|
| Description |   |
| Jira issue  |   |

#### Type of request

- [x] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: packages/jwt/src/JWTEIP191.ts [hunk 1]
```diff
@@ -72,7 +72,7 @@ export class JWT extends JwtBase implements IJWT {
       if (expirationTimestamp < Date.now()) {
         throw new Error('Expiration timestamp is in the past');
       }
-      payload.exp = Math.trunc(expirationTimestamp / 1000);
+      payload.exp = expirationTimestamp;
```
Comment (jrhender) [inline packages/jwt/src/JWTEIP191.ts:75]: @JGiter This change seems correct but I'm confused how credentials we were issuing previously had a correct expiry timestamp. Does this change require a corresponding change in callers of this code? Were callers compensating by multiplying the timestamp by 1000? Could you add some description to the PR about your understanding of this change please perhaps?

Comment (JGiter) [reply packages/jwt/src/JWTEIP191.ts:75]: @jrhender It is compensated in iam-lib here https://github.com/energywebfoundation/iam-client-lib/blob/6c74fb5c0d4185357b7678d47a5832126a9274a0/src/modules/claims/claims.service.ts#L1501. We will fix this in https://github.com/energywebfoundation/iam-client-lib/pull/633

Comment (jrhender) [reply packages/jwt/src/JWTEIP191.ts:75]: Hmm, interesting. That still doesn't explain how the timestamp is correct in previously issued JWTs as that line you linked in ICL to seems to be on the verification side. Does it make sense to you?

For example, @mhrsntrk issued this JWT yesterday using ICL and it seems to have the correct timestamp of 08/09/2023 11:22:28 (`1691569348010`)
```
eyJhbGciOiJFUzI1NksiLCJ0eXAiOiJKV1QifQ.eyJjbGFpbURhdGEiOnsicmVxdWVzdG9yRmllbGRzIjpbeyJrZXkiOiIxMjMiLCJ2YWx1ZSI6InRlc3QifV0sImNsYWltVHlwZSI6InJvbGUwMDMucm9sZXMuYXBwMDAxLmFwcHMubWhybmV4dGdlbi5pYW0uZXdjIiwiY2xhaW1UeXBlVmVyc2lvbiI6MSwiaXNzdWVyRmllbGRzIjpbXX0sImRpZCI6ImRpZDpldGhyOnZvbHRhOjB4ODU1NmM3MTRCQUI4MDFGZkI3MjkyQmQ3Mzk0MjQ2MGU3OWJlODNkRSIsInNpZ25lciI6ImRpZDpldGhyOnZvbHRhOjB4ODU1NmM3MTRCQUI4MDFGZkI3MjkyQmQ3Mzk0MjQ2MGU3OWJlODNkRSIsImNyZWRlbnRpYWxTdGF0dXMiOnsiaWQiOiJodHRwczovL2lkZW50aXR5Y2FjaGUtZGV2LmVuZXJneXdlYi5vcmcvdjEvc3RhdHVzLWxpc3QvdXJuOnV1aWQ6NmM2YmE5NWEtNDA1Mi00OGIzLTgwMTctMmNjZjMwM2Q1ODBmIiwidHlwZSI6IlN0YXR1c0xpc3QyMDIxRW50cnkiLCJzdGF0dXNMaXN0SW5kZXgiOiIwIiwic3RhdHVzTGlzdENyZWRlbnRpYWwiOiJodHRwczovL2lkZW50aXR5Y2FjaGUtZGV2LmVuZXJneXdlYi5vcmcvdjEvc3RhdHVzLWxpc3QvdXJuOnV1aWQ6NmM2YmE5NWEtNDA1Mi00OGIzLTgwMTctMmNjZjMwM2Q1ODBmIiwic3RhdHVzUHVycG9zZSI6InJldm9jYXRpb24ifSwiaXNzIjoiZGlkOmV0aHI6dm9sdGE6MHg4NTU2YzcxNEJBQjgwMUZmQjcyOTJCZDczOTQyNDYwZTc5YmU4M2RFIiwic3ViIjoiZGlkOmV0aHI6dm9sdGE6MHg4NTU2YzcxNEJBQjgwMUZmQjcyOTJCZDczOTQyNDYwZTc5YmU4M2RFIiwiZXhwIjoxNjkxNTY5MzQ4fQ.MHg4MTRiNTQzOTAwMTBmMTM4YjJjMzMwYzNkODA0NThhYmM0MTdmZDQ3ZmYzNmU3YjQzMmVkM2JkNzdhNjk2NjU2MjJhMjU0OGNkMzc4ZmI4NGUyYjA2ZmI5Nzk2YjNjMWU3MzNiY2Q5MDhmNjAwNjU2N2U3ZjI3MTI1MjcwNmZlNDFj
```

Comment (JGiter) [reply packages/jwt/src/JWTEIP191.ts:75]: It doesn't seems correct to me. The `1691569348010` is 1000 times greater then timestamp in jwt which is `1691569348`

Comment (jrhender) [reply packages/jwt/src/JWTEIP191.ts:75]: @JGiter Ah, it seems like `exp` should be in seconds, not milliseconds. So I think the current behaviour is correct and we should NOT merge this change https://stackoverflow.com/a/39926886/

**Review by jrhender — CHANGES_REQUESTED** [review]
I believe this PR should not be merged as is and the current behaviour is correct.
We should perhaps add a link to the JWT RFC documentation to avoid confusion in the future.

File: packages/jwt/src/JWTEIP191.ts (hunk 1 above)
Comment (JGiter) [reply packages/jwt/src/JWTEIP191.ts:75]: Agree. I'll revert it. Should we instead set expirationTimestamp in seconds?

Comment (jrhender) [reply packages/jwt/src/JWTEIP191.ts:75]: I don't think so, as this would be a breaking change to the library

**Pull request closed without merging**